opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -508,17 +508,39 @@ namespace Opm
     }
 
 
-
-
-
     template<typename TypeTag>
     void
     StandardWell<TypeTag>::
     assembleWellEq(const Simulator& ebosSimulator,
-                   const std::vector<Scalar>& /* B_avg */,
+                   const std::vector<Scalar>& B_avg,
                    const double dt,
                    WellState& well_state,
                    Opm::DeferredLogger& deferred_logger)
+    {
+        const bool use_inner_iterations = param_.use_inner_iterations_wells_;
+        const auto& summary_state = ebosSimulator.vanguard().summaryState();
+        const auto inj_controls = well_ecl_.isInjector() ? well_ecl_.injectionControls(summary_state) : Well::InjectionControls(0);
+        const auto prod_controls = well_ecl_.isProducer() ? well_ecl_.productionControls(summary_state) : Well::ProductionControls(0);
+        if (use_inner_iterations) {
+            Base::solveWellEqUntilConverged(ebosSimulator, B_avg, well_state, deferred_logger);
+        }
+        assembleWellEqWithoutIteration(ebosSimulator, B_avg, dt, inj_controls, prod_controls, well_state, deferred_logger);
+    }
+
+
+
+
+
+    template<typename TypeTag>
+    void
+    StandardWell<TypeTag>::
+    assembleWellEqWithoutIteration(const Simulator& ebosSimulator,
+                                   const std::vector<Scalar>& /* B_avg */,
+                                   const double dt,
+                                   const Well::InjectionControls& /*inj_controls*/,
```

Comment:
At least from this PR, we do not see the need to make this function public, and available from the  Base class. 

But it is due to the slightly different interface for the function `assembleControlEq`.  If `assembleControlEq` interface can be adjusted to the same between the two well models, you probably can use `inj_controls` and `prod_controls`, or remove them from the interface.